Octokit/Helpers/ReferenceExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Threading.Tasks;
+
+namespace Octokit.Helpers
+{
+    /// <summary>
+    /// Represents operations to simplify working with references
+    /// </summary>
+    public static class ReferenceExtensions
+    {
+        /// <summary>
+        /// Creates a branch, based off the branch specified.
+        /// </summary>
+        /// <param name="referencesClient">The <see cref="IReferencesClient" /> this method extends</param>
+        /// <param name="owner">The owner of the repository.</param>
+        /// <param name="name">The name of the repository.</param>
+        /// <param name="branchName">The new branch name</param>
+        /// <param name="baseReference">The <see cref="Reference" /> to base the branch from</param>
+        public static async Task<Reference> CreateBranch(this IReferencesClient referencesClient, string owner, string name, string branchName, Reference baseReference)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(name, "name");
+            Ensure.ArgumentNotNullOrEmptyString(branchName, "branchName");
+            Ensure.ArgumentNotNull(baseReference, "baseReference");
+
+            return await referencesClient.Create(owner, name, new NewReference("refs/heads/" + branchName, baseReference.Object.Sha));
```

Comment:
Should we validate that `branchName` doesn't already start with `refs/heads/`? I could imagine that might be a point of confusion in calling this overload and a nice exception message would be helpful. Then again, what if they really wanted the ref to be named `refs/heads/refs/heads`. I mean, that's probably silly sauce and we should just tell them to call the original `Create` method in that case.